--- a/src/altlib.h
+++ b/src/altlib.h
@@ -85,7 +85,7 @@
 extern int ncbl2_getliba(unsigned char *, int, char *, int, fseek_t *, int *,
 	    struct lmf_str *, long *);
 extern void ncbl2_ranlib(char *, int, fseek_t, char *, struct lmf_str *);
-void ncbl2_closelib();
+void ncbl2_closelib(struct lmf_str *);
 #endif
 
 #ifdef MYSQL_DB
--- a/src/mm_file.h
+++ b/src/mm_file.h
@@ -58,7 +58,7 @@
 #include "defs.h"
 #endif
 
-extern unsigned long adler32();
+extern unsigned long adler32(unsigned long, const unsigned char *, unsigned int);
 
 struct lmf_str {
   FILE *libf;		/* sequence file being read */
--- a/src/mmgetaa.c
+++ b/src/mmgetaa.c
@@ -81,16 +81,18 @@
 
 /* mmap()ed functions */
 #ifdef USE_MMAP
-int agetlibm(); void aranlibm();
-int lgetlibm(); void lranlibm();
-void vranlibm();
-int gcg_getlibm();
+int agetlibm(unsigned char *, int, char *, int, fseek_t *, int *, struct lmf_str *, long *);
+void aranlibm(char *, int, fseek_t, char *, struct lmf_str *);
+int lgetlibm(unsigned char *, int, char *, int, fseek_t *, int *, struct lmf_str *, long *);
+void lranlibm(char *, int, fseek_t, char *, struct lmf_str *);
+void vranlibm(char *, int, fseek_t, char *, struct lmf_str *);
+int gcg_getlibm(unsigned char *, int, char *, int, fseek_t *, int *, struct lmf_str *, long *);
 
-int (*getlibam[])()={
+int (*getlibam[])(unsigned char *, int, char *, int, fseek_t *, int *, struct lmf_str *, long *)={
   agetlibm,lgetlibm, NULL, NULL,NULL,agetlibm,gcg_getlibm
 };
 
-void (*ranlibam[])()={
+void (*ranlibam[])(char *, int, fseek_t, char *, struct lmf_str *)={
   aranlibm,lranlibm,NULL,NULL,NULL,vranlibm,vranlibm
 };
 #endif
